### PR TITLE
8286908: ECDSA signature should not return parameters

### PR DESCRIPTION
**Pull request by wangweij**
Let ECDSA's `engineGetParameters()` always return null. At the same time, remove the remembered `sigParams` field. One behavior change is that after calling `setParameter()`, one can call `init()` again with a key using different parameters. I think this should be allowed since we are reusing the signature object with a brand new key.

`setParameter` is kept unchanged to be able to deal with certificates still having parameters after the signature algorithm object identifier. See https://bugs.openjdk.java.net/browse/JDK-8225745.

Also added SHA1withECDSA to the no-NULL list in `KnownOIDs`.

All security-related tests passed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issues
 * [JDK-8286908](https://bugs.openjdk.java.net/browse/JDK-8286908): ECDSA signature should not return parameters
 * [JDK-8286982](https://bugs.openjdk.java.net/browse/JDK-8286982): ECDSA signature should not return parameters (**CSR**)

### Reviewers
 * [Anthony Scarpino](https://openjdk.java.net/census#ascarpino) (@ascarpino - **Reviewer**) ⚠️ Review applies to [b13de5a9](https://git.openjdk.java.net/jdk/pull/8758/files/b13de5a91a71d2054e560f2fff2b268c2fd340dd)
 * [Hai-May Chao](https://openjdk.java.net/census#hchao) (@haimaychao - Committer) ⚠️ Review applies to [b13de5a9](https://git.openjdk.java.net/jdk/pull/8758/files/b13de5a91a71d2054e560f2fff2b268c2fd340dd)
 * [Valerie Peng](https://openjdk.java.net/census#valeriep) (@valeriepeng - **Reviewer**) ⚠️ Review applies to [aceeb4a7](https://git.openjdk.java.net/jdk/pull/8758/files/aceeb4a74c32f364d318bedd840dbeb395a32a18)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8758/head:pull/8758` \
`$ git checkout pull/8758`

Update a local copy of the PR: \
`$ git checkout pull/8758` \
`$ git pull https://git.openjdk.java.net/jdk pull/8758/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8758`

View PR using the GUI difftool: \
`$ git pr show -t 8758`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8758.diff">https://git.openjdk.java.net/jdk/pull/8758.diff</a>

</details>

### CODE REVIEW

**Comment by jnimeh** [comment]
Do the behavioral changes you've cited in the PR description warrant a CSR, or do you feel this behavioral change is still consistent with the current Signature API documentation?

**Comment by wangweij** [comment]
> Do the behavioral changes you've cited in the PR description warrant a CSR, or do you feel this behavioral change is still consistent with the current Signature API documentation?

I think so. In fact, after this change, there's simply no parameters for the signature, so calling `setParameter` should not change any internal state and thus has no effect on other calls.

**Comment by ascarpino** [comment]
Looks good to me.
The changes seem within the spec to me.

**Review by ascarpino — APPROVED** [review]

**Review by haimaychao — APPROVED** [review]

**Review by valeriepeng — APPROVED** [review]
Looks fine.

File: src/java.base/share/classes/sun/security/util/SignatureUtil.java [hunk 1]
```diff
@@ -151,17 +151,9 @@ public static AlgorithmParameterSpec getParamSpec(String sigName,
...
-                    throw new ProviderException("Error handling EC parameters", e);
-                }
-                // ECUtil discards exception and returns null, so we need to check
-                // the returned value
-                if (paramSpec == null) {
-                    throw new ProviderException("Error handling EC parameters");
-                }
+                // Some certificates have params in an ECDSA algorithmID,
+                // they are useless and should be ignored.
+                return null;
```
Comment (haimaychao) [inline src/java.base/share/classes/sun/security/util/SignatureUtil.java:156]: Suggest adding the RFC reference for this change.

**Comment by haimaychao** [comment]
CSR and updated webrev look good.

**Pull request closed without merging**